### PR TITLE
Destroy open streams on `dht.destroy()`

### PR DESCRIPTION
**Pull request by kasperisager**

### CODE REVIEW

**Comment by kasperisager** [comment]
@mafintosh This is ready for review.

File: lib/raw-stream-set.js [hunk 1]
```diff
@@ -26,4 +26,8 @@ module.exports = class RawStreamSet {
       self._streams.delete(id)
     }
   }
+
+  async destroy () {
+    for (const stream of this._streams.values()) stream.destroy()
```
Comment (kasperisager) [inline lib/raw-stream-set.js:31]: Not sure if this should maybe wait for the streams to emit `close`?

Comment (mafintosh) [reply lib/raw-stream-set.js:31]: I think it should

Comment (kasperisager) [reply lib/raw-stream-set.js]: @mafintosh Should be all set now.

**Comment by kasperisager** [comment]
@mafintosh This is ready for another look.

**Comment by mafintosh** [comment]
Looks good. Test failure is due to a test not error handling i think (cause we are tearing down the streams aggresively now on test end)

**Comment by LuKks** [comment]
Great, it also fixes https://github.com/hyperswarm/dht/issues/86
Only difference with v5 and v6 is that now it's giving `ECONNRESET` but that is correct as the socket was open

File: test/connections.js [hunk 2]
```diff
@@ -408,6 +408,7 @@ test('relayed connection', async function (t) {
   const socket = b.connect(server.address().publicKey)
 
   server.on('connection', (socket) => {
+    socket.on('error', () => {})
```
Comment (LuKks) [inline test/connections.js:411]: Sorry, have a question, why would this test give error? Both sides are gracefully doing `socket.end()`

Comment (kasperisager) [reply test/connections.js:411]: The client side doesn't and is now forcefully destroyed on teardown.

**Comment by mafintosh** [comment]
6.3.5

**Pull request merged**